io-engine/src/subsys/config/opts.rs

```diff
@@ -279,25 +284,29 @@ impl Default for NvmfTcpTransportOpts {
 /// know about it during compile time.
 impl From<NvmfTcpTransportOpts> for spdk_nvmf_transport_opts {
     fn from(o: NvmfTcpTransportOpts) -> Self {
-        Self {
-            max_queue_depth: o.max_queue_depth,
-            max_qpairs_per_ctrlr: o.max_qpairs_per_ctrl,
-            in_capsule_data_size: o.in_capsule_data_size,
-            max_io_size: o.max_io_size,
-            io_unit_size: o.io_unit_size,
-            max_aq_depth: o.max_aq_depth,
-            num_shared_buffers: o.num_shared_buf,
-            buf_cache_size: o.buf_cache_size,
-            dif_insert_or_strip: o.dif_insert_or_strip,
-            reserved29: Default::default(),
-            abort_timeout_sec: o.abort_timeout_sec,
-            association_timeout: 120000,
-            transport_specific: std::ptr::null(),
-            opts_size: std::mem::size_of::<spdk_nvmf_transport_opts>() as u64,
-            acceptor_poll_rate: o.acceptor_poll_rate,
-            zcopy: o.zcopy,
-            reserved61: Default::default(),
-        }
+        struct_size_init!(
+            Self {
+                max_queue_depth: o.max_queue_depth,
+                max_qpairs_per_ctrlr: o.max_qpairs_per_ctrl,
+                in_capsule_data_size: o.in_capsule_data_size,
+                max_io_size: o.max_io_size,
+                io_unit_size: o.io_unit_size,
+                max_aq_depth: o.max_aq_depth,
+                num_shared_buffers: o.num_shared_buf,
+                buf_cache_size: o.buf_cache_size,
+                dif_insert_or_strip: o.dif_insert_or_strip,
+                reserved29: Default::default(),
+                abort_timeout_sec: o.abort_timeout_sec,
+                association_timeout: 120000,
+                transport_specific: std::ptr::null(),
+                acceptor_poll_rate: o.acceptor_poll_rate,
+                zcopy: o.zcopy,
+                reserved61: Default::default(),
+                ack_timeout: 0,
+                data_wr_pool_size: 0,
```

Comment:
these are new fields, should they be added in the `NvmfTcpTransportOpts` ?

Reply:
Done.